src/main/java/hannah/task/Deadline.java

```diff
@@ -0,0 +1,35 @@
+package hannah.task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class Deadline extends Task {
+    String name;
+    LocalDate deadline;
+    public Deadline(String name){
+        super(name);
+    }
+    @Override
+    public String toString(){
+        return("[D]" + super.toString() + " by " +  this.deadline);
+    }
+
+    @Override
+    public void setDeadline(String deadline) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+        try {
+            this.deadline = LocalDate.parse(deadline, formatter);
+        } catch (DateTimeParseException e) {
+            System.out.println("Invalid date format. Please use yyyy-MM-dd.");
+        }
+    }
+    public LocalDate getDeadline(){
+        return this.deadline;
+    }
+    @Override
+    public String saveTaskFormat(){
+        String line = "D | " + (isDone ? "1" : "0") + " | " + super.name + " | by " + this.deadline;
+        return line + System.lineSeparator();
+    }
```

Comment:
Not a coding standard issue, perhaps you will want to separate line 26 to 27 and line 29 to 30.